apps/sparo-lib/src/services/test/GitRemoteFetchConfigService.test.ts

```diff
@@ -0,0 +1,28 @@
+import { getFromContainer } from '../../di/container';
+import { GitRemoteFetchConfigService } from '../GitRemoteFetchConfigService';
+
+describe(GitRemoteFetchConfigService.name, () => {
+  const gitRemoteFetchConfigService = getFromContainer(GitRemoteFetchConfigService);
+
+  describe(gitRemoteFetchConfigService.getBranchesInfoFromRemoteFetchConfig, () => {
+    it('should work', () => {
+      const values: string[] = [
+        '+refs/heads/*:refs/remotes/origin/*',
+        '+refs/heads/release:refs/remotes/origin/release',
+        '+refs/heads/feat/abc:refs/remotes/origin/feat/abc'
+      ];
+
+      const expectedContaining: Record<string, string> = {
+        '*': values[0],
+        release: values[1],
+        'feat/abc': values[2]
```

Comment:
It would be interesting to consider what happens if someone uses the `sparo` CLI to provide a branch or remote name containing punctuation.

IIRC the Git branch name is allowed to contain ` ' `, ` " `.  But it should not contain most shell characters, nor should it contain `\n` or `:` etc.

According to our goals, with respect to these concerns of security vulnerabilities or getting into a corrupted state, Sparo is [only expected](https://tiktok.github.io/sparo/pages/reference/security/) to handle it as well as Git does.

Reply:
The tested function `getBranchesInfoFromRemoteFetchConfig` reads the inputs from the git configuration. Which means they are most likely set up by other git commands. That says that we can somehow believe the input won't contains disallowed punctuation.

Are you talking about to add a verification for the branch names?

----
Appendix:

Allowed Punctuation in Branch Names

	•	Period (.)
	•	Comma (,)
	•	Dash (-)
	•	Underscore (_)
	•	Slash (/)

Disallowed Punctuation in Branch Names

	•	Space ( )
	•	Tilde (~)
	•	Caret (^)
	•	Colon (:)
	•	Question Mark (?)
	•	Asterisk (*)
	•	Open Bracket ([)
	•	Backslash (\)

Reply:
We can reasonably trust inputs received from Git's config files. I was thinking more that where these branch names are obtained from CLI parameters, maybe we should apply a RegEx to make sure they don't contain any forbidden characters, if that string will later end up as a `spawn` argument.  `Executable.spawn()` will try hard to escape weird characters, but if we're sure such characters are wrong, it might be worth validating.  You don't need to address the question in this PR though, just keep [SS3](https://tiktok.github.io/sparo/pages/reference/security/#ss3-git-parameters-may-include-special-characters) in mind.